liquigraph-core/src/test/java/org/liquigraph/core/io/lock/LockableConnectionTest.java

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright 2014-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.liquigraph.core.io.lock;
+
+import org.junit.Test;
+
+import java.sql.Connection;
+
+import static org.mockito.Mockito.inOrder;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class LockableConnectionTest {
+
+    @Test
+    public void disables_auto_commit() throws Exception {
+        Connection delegate = mock(Connection.class);
+
+        new LockableConnection(delegate, new LiquigraphLock());
+
+        verify(delegate).setAutoCommit(false);
+    }
+
+    @Test
+    public void restores_auto_commit_upon_close() throws Exception {
+        Connection delegate = mock(Connection.class);
+        when(delegate.getAutoCommit()).thenReturn(true);
+
+        LockableConnection lockableConnection = new LockableConnection(delegate, new LiquigraphLock());
+
+        inOrder(delegate).verify(delegate).setAutoCommit(false);
```

Comment:
With a fresher eye in the morning, and according to the [Mockito examples](http://site.mockito.org/mockito/docs/current/org/mockito/Mockito.html#6), the same `InOrder` object should be used for all the verifications. Otherwise it doesn't actually test the order, since all the verifications are independent.

You could also skip this particular verification, since it's already asserted in the other test.